File: examples/http-upload.lua
local HTTP = require("http")
local Utils = require("utils")
local Table = require("table")

HTTP.create_server("0.0.0.0", 8080, function (req, res)
  p("on_request", req)
  local chunks = {}
  local length = 0
  req:on('data', function (chunk, len)
    p("on_data", {len=len})
    length = length + 1
    chunks[length] = chunk
  end)
  req:on('end', function ()
    local body = Table.concat(chunks, "")
    p("on_end", {total_len=#body})
    body = "length = " .. tostring(#body) .. "\n"
    res:write_head(200, {
      ["Content-Type"] = "text/plain",
      ["Content-Length"] = #body
    })
    res:finish(body)
  end)
    
end)

print("Server listening at http://localhost:8080/")

File: examples/broken/http-upload.js
var HTTP = require("http");

HTTP.createServer(function (req, res) {
  var length = 0
  var total = 0
  req.on('data', function (chunk) {
    length++;
    total += Buffer.byteLength(chunk)
  });
  req.on('end', function () {
    var body = "length = " + total + "\n";
    res.writeHead(200, {
      "Content-Type": "text/plain",
//      "Content-Length": Buffer.byteLength(body)
    });
    res.end(body);
  });
    
}).listen(8080)

console.log("Server listening at http://localhost:8080/")

